geode-apis-compatible-with-redis/src/main/java/org/apache/geode/redis/internal/executor/sortedset/ZStoreExecutor.java

```diff
@@ -0,0 +1,133 @@
+import org.apache.geode.redis.internal.executor.AbstractExecutor;
+import org.apache.geode.redis.internal.executor.RedisResponse;
+import org.apache.geode.redis.internal.netty.Coder;
+import org.apache.geode.redis.internal.netty.Command;
+import org.apache.geode.redis.internal.netty.ExecutionHandlerContext;
+
+public abstract class ZStoreExecutor extends AbstractExecutor {
+
+  @Override
+  public RedisResponse executeCommand(Command command, ExecutionHandlerContext context) {
+    List<byte[]> commandElements = command.getProcessedCommand();
+    ListIterator<byte[]> argIterator = commandElements.listIterator();
+    RedisResponse syntaxErrorResponse = RedisResponse.error(ERROR_SYNTAX);
+
+    // Skip command and destination key
+    argIterator.next();
+    argIterator.next();
+
+    // get the number of keys
+    int numKeys;
+    try {
+      numKeys = narrowLongToInt(Coder.bytesToLong(argIterator.next()));
+      if (numKeys > commandElements.size() - 2) {
+        return syntaxErrorResponse;
+      } else if (numKeys <= 0) {
+        return RedisResponse.error(ERROR_KEY_REQUIRED);
+      }
+    } catch (NumberFormatException ex) {
+      return syntaxErrorResponse;
+    }
+
+    List<ZKeyWeight> keyWeights = new ArrayList<>(numKeys);
+    ZAggregator aggregator = ZAggregator.SUM;
+    byte[] argument;
+
+    // get all the keys
+    for (int i = 0; i < numKeys; i++) {
+      if (!argIterator.hasNext()) {
+        return syntaxErrorResponse;
+      }
+      argument = argIterator.next();
+      if (Arrays.equals(toUpperCaseBytes(argument), bWEIGHTS)
+          || Arrays.equals(toUpperCaseBytes(argument), bAGGREGATE)) {
+        return syntaxErrorResponse;
+      }
+      keyWeights.add(new ZKeyWeight(new RedisKey(argument), 1D));
+    }
+
+    while (argIterator.hasNext()) {
+      argument = argIterator.next();
+      // found AGGREGATE keyword; parse aggregate
+      if (Arrays.equals(toUpperCaseBytes(argument), bAGGREGATE)) {
+        if (!argIterator.hasNext()) {
+          return syntaxErrorResponse;
+        }
+        argument = argIterator.next();
+        if (Arrays.equals(toUpperCaseBytes(argument), bWEIGHTS)) {
+          return syntaxErrorResponse; // there must be an aggregate between 'AGGREGATE' & 'WEIGHTS'
+        }
+        try {
+          aggregator = ZAggregator.valueOf(Coder.bytesToString(toUpperCaseBytes(argument)));
+        } catch (IllegalArgumentException e) {
+          return syntaxErrorResponse;
+        }
+        // found WEIGHTS keyword; parse weights
+      } else if (Arrays.equals(toUpperCaseBytes(argument), bWEIGHTS)) {
+        for (int i = 0; i < numKeys; i++) {
+          if (!argIterator.hasNext()) {
+            return syntaxErrorResponse;
+          }
+          argument = argIterator.next();
+          if (Arrays.equals(toUpperCaseBytes(argument), bAGGREGATE)) {
+            return syntaxErrorResponse; // there must be # weights between 'WEIGHTS' & 'AGGREGATE'
+          }
```

Comment:
This behaviour doesn't match native Redis. The below test fails with Radish but passes with native Redis:
```
  @Test
  public void shouldReturnWeightNotAValidFloat_givenWeightsFollowedByCorrectNumberOfArgumentsIncludingAggregate() {
    assertThatThrownBy(
        () -> jedis.sendCommand(NEW_SET, Protocol.Command.ZINTERSTORE, NEW_SET, "3",
            KEY1, KEY2, KEY3, "WEIGHTS", "1", "AGGREGATE", "SUM"))
                .hasMessage("ERR " + RedisConstants.ERROR_WEIGHT_NOT_A_FLOAT);
  }
```
I think that this check is not needed, because we only care if there are enough weights specified (if there aren't, `argIterator.hasNext()` returns false and we return a syntax error) and that those weights are valid floats (if they're not, we get a `NumberFormatException` below and return `ERROR_WEIGHT_NOT_A_FLOAT`).